WordPress/gutenberg
# Remove `gutenberg_should_block_use_interactivity_api ` and the old implementation of interactive blocks

## What?
Remove the `gutenberg_should_block_use_interactivity_api ` function and the conditionals using it. Additionally, I took the opportunity to remove the old implementation of blocks using the Interactivity API.

## Why?
Now that we are planning on using the Interactivity API in both WordPress 6.4 and Gutenberg, we don't need to keep supporting the old versions.

## How?
I just removed the function that was checking if it was in Gutenberg and I removed the conditionals to ensure they work the same way in WordPress and Gutenberg.

Additionally, I reverted the changes made [here](https://github.com/WordPress/gutenberg/pull/52553/files#diff-a1f260292faa29d2f30136c99f839fd77d716b21450ec6ab89bb11e6f721a1eb) to only have the Interactivity API version.

## Testing Instructions
* Include a navigation block, check that it only enqueues the `view.js` file with the Interactivity API code, and check that the overlay menu and the submenu work as expected.
* Include a file block, check it only enqueues the `view.js` file with the Interactivity API, and check it shows the preview when the setting is activated.
